simple/app.go

```diff
@@ -88,19 +132,37 @@ type AppManagedKind struct {
 	Watcher operator.ResourceWatcher
 	// Validator is an optional ValidatingAdmissionController for the Kind. It will be run only for validation
 	// of this specific version.
-	Validator resource.ValidatingAdmissionController
+	Validator KindValidator
 	// Mutator is an optional MutatingAdmissionController for the Kind. It will be run only for mutation
 	// of this specific version.
-	Mutator resource.MutatingAdmissionController
-	// SubresourceRoutes are an optional map of subresource paths to a route handler.
+	Mutator KindMutator
+	// CustomRoutes are an optional map of subresource paths to a route handler.
 	// If supported by the runner, calls to these subresources on this particular version will call this handler.
-	SubresourceRoutes map[string]func(ctx context.Context, writer http.ResponseWriter, req *app.SubresourceRequest) error
+	CustomRoutes []AppCustomRouteHandler
```

Comment:
That makes sense to me, let me implement this

Reply:
So it looks like using a map doesn't forbid duplicate entries, so I still have the duplicates check. Given this, does having this be a slice of 
```type AppCustomRoute struct {
    Method AppCustomRouteMethod
    Path string
    Handler AppCustomRouteHandler
}```
Make more sense? I'll commit the version with the map for now.

Reply:
> So it looks like using a map doesn't forbid duplicate entries as a compiler error, but instead just overwrites them.

Good point, but at least in this case it will be on the developer. Assuming that they declare the map in-line, the compiler will fail, since it can statically determine duplicate entries. If the developer builds the route map dynamically they will still have the ability to shoot themselves in the foot but to prevent that we could in the future change the custom type to something that uses the map internally but doesn't expose the map directly, instead having methods like `Add` / `Delete`.

WDYT?

Reply:
> Assuming that they declare the map in-line, the compiler will fail, since it can statically determine duplicate entries
That's what I mean--it doesn't run into a compile-time error, the map keys just overwrite each other even when in-line ([playground example](https://go.dev/play/p/Yar8q1QYMHM)). Not sure if this is desirable UX, as a user may think (like I did initially) that if they in-line declare the map, they're protected from duplicates by the compiler.

Reply:
Oh, I see now, must be that it only works for scalar types. Anyhow, I think it should be OK for now – like I said, we can introduce a new custom wrapper over this map in later PRs.